### PR TITLE
Add terraform for Databricks and ADF

### PR DESCRIPTION
**Pull request by tanya-borisova**
Added Terraform configuration for Databricks and ADF. 

Verified by creating a job on ADF using Databricks and checking that it worked (meaning that cluster configuration and identity setup should be correct).

### CODE REVIEW

File: infrastructure/transform/main.tf [hunk 1]
```diff
@@ -12,11 +12,66 @@
...
+  custom_tags = {
+    "ResourceClass" = "SingleNode"
+  }
+
+  depends_on = [azurerm_databricks_workspace.databricks]
+}
+
+resource "azurerm_role_assignment" "adf_can_create_clusters" {
+  scope                = azurerm_databricks_workspace.databricks.id
+  role_definition_name = "Contributor"
```
Comment (tanya-borisova) [inline infrastructure/transform/main.tf:55]: Maybe there is a more granular role I could use, but didn't manage to find it

File: infrastructure/terragrunt.hcl [hunk 2]
```diff
@@ -32,6 +32,10 @@ terraform {
       source  = "hashicorp/azurerm"
       version = ">= 3.32"
     }
+    databricks = {
+      source = "databricks/databricks"
+      version = "1.9.0"
```
Comment (t-young31) [inline infrastructure/terragrunt.hcl:37]: 1.9.1 is latest? is there a reason not to use that?

File: infrastructure/transform/main.tf [hunk 3]
```diff
@@ -12,11 +12,66 @@
...
+  depends_on = [azurerm_databricks_workspace.databricks]
+}
+
+data "databricks_node_type" "smallest" {
+  # Providing no required configuration, Databricks will pick the smallest node possible
+  depends_on = [azurerm_databricks_workspace.databricks]
+}
+
+resource "databricks_cluster" "fixed_single_node" {
+  cluster_name            = "Fixed Job Cluster"
```
Comment (t-young31) [inline infrastructure/transform/main.tf:36]: any potential problems with spaces in names?

File: infrastructure/transform/main.tf [hunk 4]
```diff
@@ -12,11 +12,66 @@
...
+}
+
+resource "azurerm_role_assignment" "adf_can_create_clusters" {
+  scope                = azurerm_databricks_workspace.databricks.id
+  role_definition_name = "Contributor"
+  principal_id         = azurerm_data_factory.adf.identity[0].principal_id
+}
+
+resource "azurerm_data_factory" "adf" {
+  name                = "${var.prefix}-${var.environment}-dbks"
```
Comment (t-young31) [inline infrastructure/transform/main.tf:60]: `adf` suffix might be more appropriate(?)

**Review by t-young31 — APPROVED** [review]
Looks great to me – thanks 🚀  (not tried to deploy myself)

File: infrastructure/transform/main.tf (hunk 4 above)
Comment (tanya-borisova) [reply infrastructure/transform/main.tf:60]: Good spot! I got too excited with copy paste

File: infrastructure/transform/main.tf (hunk 3 above)
Comment (tanya-borisova) [reply infrastructure/transform/main.tf:36]: Didn't seem to cause any problems, and also default clusters that Databricks creates have spaces in their names so should be fine

File: infrastructure/terragrunt.hcl (hunk 2 above)
Comment (tanya-borisova) [reply infrastructure/terragrunt.hcl:37]: Good point, will update

**Pull request merged**